ompi/mca/coll/tuned/coll_tuned_decision_fixed.c

```diff
@@ -806,8 +806,10 @@ int ompi_coll_tuned_scatter_intra_dec_fixed(const void *sbuf, int scount,
                                                      root, comm, module);
     } else if ((communicator_size < ompi_coll_tuned_scatter_min_procs) &&
                (communicator_size > intermediate_comm_size) &&
-               (block_size >= ompi_coll_tuned_scatter_intermediate_msg) &&
-               (block_size < ompi_coll_tuned_scatter_large_msg)) {
+               (ompi_coll_tuned_scatter_intermediate_msg > -1) &&
+               (block_size >= (size_t) ompi_coll_tuned_scatter_intermediate_msg) &&
+               (ompi_coll_tuned_scatter_large_msg > -1) &&
+               (block_size < (size_t) ompi_coll_tuned_scatter_large_msg)) {
         return ompi_coll_base_scatter_intra_linear_nb(sbuf, scount, sdtype,
                                                       rbuf, rcount, rdtype,
                                                       root, comm, module,
```

Comment:
It makes little sense for these 2 cut-of MCA params (ompi_coll_tuned_scatter_intermediate_msg and ompi_coll_tuned_scatter_large_msg) to be negative. Maybe we should fix those.

Reply:
I agree.

Changing them both to be a `size_t` or an `unsigned` seems to make sense. That way if the user does specify -1 at the command line, it would essentially force this path.